src/docker/registry.nim

```diff
@@ -60,25 +91,38 @@ iterator iterDaemonRegistryConfigs(self: DockerImage): RegistryConfig =
   if cert != "":
     trace("docker: found CA certificate for " & self.registry & " at " & path & " in docker daemon")
     yield RegistryConfig(
-      scheme:     "https://",
-      verifyMode: CVerifyPeer,
-      certPath:   path,
-      pinnedCert: writeNewTempFile(
+      scheme:      "https://",
+      registry:    self.registry,
+      prefix:      prefix,
+      verifyMode:  CVerifyPeer,
+      fallthrough: fallthrough,
+      certPath:    path,
+      pinnedCert:  writeNewTempFile(
         cert,
         prefix = self.domain,
         suffix = ".crt",
       ),
     )
 
-  template yieldInsecure() =
-    trace("docker: " & self.registry & " will attempt TLS without verifying server cert")
-    yield RegistryConfig(scheme: "https://", verifyMode: CVerifyNone)
-    yield RegistryConfig(scheme: "http://", verifyMode: CVerifyNone)
-
   for i in getDockerInfoSubList("insecure registries:"):
     if self.registry == i or self.domain == i:
       trace("docker: " & i & " is configured as an insecure registry in docker daemon")
-      yieldInsecure()
+      trace("docker: " & self.registry & " will attempt TLS without verifying server cert")
```

Comment:
does nim have something similar to __module__ or __file__ so we can avoid manually putting the `docker:` prefix in?

Reply:
this log function is from con4m, not nim builtin. in there we have to provide full log string all the time